### ISSUE TITLE
Validation Set for Tuning

### ISSUE THREAD

**Issue by tgroechel** [interaction-lab/BehaviorTreeFromDemonstration#47]
Unsure if we should be doing some hyperparam tuning or not. We have K-Fold cross with cost_complexity_pruning (see #39 ) to choose a best alpha but not sure if we should look at other hyperparams like max_depth etc
